webview/icons/generate.mjs

```diff
@@ -0,0 +1,103 @@
+import { transform } from '@svgr/core'
+import {
+  appendFile,
+  mkdir,
+  readdir,
+  readFile,
+  rm,
+  writeFile
+} from 'node:fs/promises'
+import path from 'path'
+
+const iconsPath = 'src/shared/components/icons/'
+
+try {
+  await rm(iconsPath, { recursive: true, force: true })
+  await mkdir(iconsPath)
+} catch (err) {
+  console.error(err.message)
+}
+
+const customIcons = []
+try {
+  const files = await readdir('icons')
+  for (const file of files) {
+    if (path.extname(file) === '.svg') {
+      customIcons.push(`icons/${file}`)
+    }
+  }
+} catch (err) {
+  console.error(err.message)
+}
+
+const codiconsPath = '../node_modules/@vscode/codicons/src/icons'
+const codicons = [
```

Comment:
Since this is a constant that will be added to/deleted from to create icons, should we place it in a separate constants file?

Reply:
That is a great idea. Will do.